packages/askui-nodejs/src/utils/http/http-client-got.ts

```diff
@@ -15,45 +15,68 @@ export class HttpClientGot {
     readonly token?: string,
     readonly customHeaders?: Record<string, string>,
     private readonly cookies: Record<string, string> = {},
-    readonly proxyAgents?: { http: http.Agent, https: https.Agent },
+    readonly proxyAgents?: { http: http.Agent; https: https.Agent },
   ) {
     this.initHeaders(token, customHeaders);
     this.askuiGot = got.extend(proxyAgents ? { agent: proxyAgents } : {});
   }
 
-  private initHeaders(token?: string, customHeaders: Record<string, string> = {}) {
+  private initHeaders(
+    token?: string,
+    customHeaders: Record<string, string> = {},
+  ) {
     const credentials = token ? new Credentials(token) : undefined;
     this.headers = {
-      ...(credentials ? { Authorization: `Basic ${credentials?.base64Encoded}` } : {}),
+      ...(credentials
+        ? { Authorization: `Basic ${credentials?.base64Encoded}` }
+        : {}),
       ...customHeaders,
     };
   }
 
-  private injectHeadersAndCookies(url: string, options: OptionsOfJSONResponseBody) {
+  private injectHeadersAndCookies(
+    url: string,
+    options: OptionsOfJSONResponseBody,
+  ) {
     const cookieJar = new CookieJar();
-    Object.keys(this.cookies).map((key) => `${key}=${this.cookies[key]}`).forEach((cookie) => {
-      cookieJar.setCookieSync(cookie, url);
-    });
+    Object.keys(this.cookies)
+      .map((key) => `${key}=${this.cookies[key]}`)
+      .forEach((cookie) => {
+        cookieJar.setCookieSync(cookie, url);
+      });
     return { ...options, headers: this.headers, cookieJar };
   }
 
-  async post<T>(url: string, data: Record<string | number | symbol, unknown>): Promise<T> {
-    const options = this.injectHeadersAndCookies(url, { json: data, responseType: 'json', throwHttpErrors: false });
-    const { body, statusCode, headers } = await this.askuiGot.post<T>(url, options);
+  async post<T>(
+    url: string,
+    data: Record<string | number | symbol, unknown>,
+  ): Promise<{ headers: http.IncomingHttpHeaders; body: T }> {
+    const options = this.injectHeadersAndCookies(url, {
+      json: data,
+      responseType: 'json',
+      throwHttpErrors: false,
+    });
+    const { body, statusCode, headers } = await this.askuiGot.post<T>(
+      url,
+      options,
+    );
     if (headers['deprecation'] !== undefined) {
       logger.warn(headers['deprecation']);
     }
```

Comment:
@adi-wan-askui  Did you saw this? We already check here the header.

Reply:
We could move this to the `inference-client.ts` too.

Reply:
I decided against it because the `deprecation` header is a generic header that is generally applicable, see https://greenbytes.de/tech/webdav/draft-ietf-httpapi-deprecation-header-latest.html (although we currently don't use it as specified there).